pyartifactory/objects/artifact.py

```diff
@@ -90,9 +96,29 @@ def deploy(self, local_file_location: Union[Path, str], artifact_path: Union[Pat
                 for file in files:
                     self.deploy(Path(f"{root}/{file}"), Path(f"{new_root}/{file}"))
         else:
-            with local_file.open(mode="rb") as streamfile:
-                self._put(route=artifact_folder.as_posix(), data=streamfile)
-                logger.debug("Artifact %s successfully deployed", artifact_folder.name)
+            if checksums:
+                # based on https://github.com/thenewnano/python-artifactory/commit/3c095f88f5e4153d1feffbf3769d038eb16b4016
+                artifact_check_sums = Checksums.generate(local_file)
+                headers = {
+                    "X-Checksum-Deploy": "true",
+                    "X-Checksum-Sha1": artifact_check_sums.sha1,
+                    "X-Checksum-Sha256": artifact_check_sums.sha256,
+                    "X-Checksum": artifact_check_sums.md5,
+                }
+            else:
+                headers = {}
+            try:
+                self._put(f"{artifact_path}", headers=headers)
+            except requests.exceptions.HTTPError as error:
+                if error.response.status_code == 404 and not checksums:
```

Comment:
What would happen if checksums=True ?

I believe we should keep it simple and do something like this:

```
if not checksums:
  headers["X-Checksum-Deploy"] = "false"
  with local_file.open("rb") as stream:
      self._put(f"{artifact_path}", data=stream, headers=headers)
else:
  self._put(f"{artifact_path}", headers=headers)

logger.debug("Artifact %s successfully deployed", local_file)
```